calendar-bundle/src/Resources/contao/modules/ModuleEventReader.php

```diff
@@ -287,29 +290,14 @@ protected function compile()
 		$objTemplate->addBefore = false;
 
 		// Add an image
-		if ($objEvent->addImage && $objEvent->singleSRC)
+		if ($objEvent->addImage && null !== ($figureBuilder = $this->getFigureBuilderIfResourceExists($objEvent->singleSRC)))
 		{
-			$objModel = FilesModel::findByUuid($objEvent->singleSRC);
-
-			if ($objModel !== null && is_file(System::getContainer()->getParameter('kernel.project_dir') . '/' . $objModel->path))
-			{
-				// Do not override the field now that we have a model registry (see #6303)
-				$arrEvent = $objEvent->row();
-
-				// Override the default image size
-				if ($this->imgSize)
-				{
-					$size = StringUtil::deserialize($this->imgSize);
-
-					if ($size[0] > 0 || $size[1] > 0 || is_numeric($size[2]) || ($size[2][0] ?? null) === '_')
-					{
-						$arrEvent['size'] = $this->imgSize;
-					}
-				}
-
-				$arrEvent['singleSRC'] = $objModel->path;
-				$this->addImageToTemplate($objTemplate, $arrEvent, null, null, $objModel);
-			}
+			$figureBuilder
+				->setSize($objEvent->size ?: $this->imgSize ?: null)
```

Comment:
This now prefers `$objEvent->size` whereas the old code preferred `$this->imgSize` if it was set to a non-empty size.

Reply:
Fixed in 1db55b3d80534f519efddf28902881709787bf51